jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/AwsConnectionManager.kt

```diff
@@ -137,49 +131,61 @@ abstract class AwsConnectionManager(private val project: Project) : SimpleModifi
 
     @Synchronized
     private fun changeFieldsAndNotify(fieldUpdateBlock: () -> Unit) {
-        incModificationCount()
-
-        validationJob?.cancel(CancellationException("Newer connection settings chosen"))
         val isInitial = connectionState is ConnectionState.InitializingToolkit
         connectionState = ConnectionState.ValidatingConnection
 
+        // Grab the current state stamp
+        val modificationStamp = this.modificationCount
+
         fieldUpdateBlock()
 
-        validationJob = GlobalScope.launch(Dispatchers.IO) {
-            val credentialsIdentifier = selectedCredentialIdentifier
-            val region = selectedRegion
+        val validateCredentialsResult = validateCredentials(selectedCredentialIdentifier, selectedRegion, isInitial)
+        validationJob.getAndSet(validateCredentialsResult)?.cancel()
+
+        validateCredentialsResult.onSuccess {
+            // Validate we are still operating in the latest view of the world
+            if (modificationStamp == this.modificationCount) {
+                connectionState = it
+            } else {
+                LOGGER.warn("validateCredentials returned but the account manager state has been manipulated before results were back, ignoring")
+            }
+        }
+    }
+
+    private fun validateCredentials(credentialsIdentifier: CredentialIdentifier?, region: AwsRegion?, isInitial: Boolean): AsyncPromise<ConnectionState> {
+        val promise = AsyncPromise<ConnectionState>()
+        ApplicationManager.getApplication().executeOnPooledThread {
             if (credentialsIdentifier == null || region == null) {
-                connectionState = ConnectionState.IncompleteConfiguration(credentialsIdentifier, region)
-                incModificationCount()
-                return@launch
+                promise.setResult(ConnectionState.IncompleteConfiguration(credentialsIdentifier, region))
```

Comment:
Can you try this in Rider?

Reply:
There is no issue with it, this process.onSuccess doesnt need to be on EDT